### PR TITLE
BaaS

### PR DESCRIPTION
**Pull request by kadenbking**
@kenancasey I've merged all current work into dev. You're welcome to leave any comments you want added to dev before we merge into main here. I went ahead and changed a couple of small things since you looked at it yesterday:

- Auto fill mode works correctly now
- Users can only be in 5 groups (added more alerts, restraints around that)
- Changed the group / group leaderboard page title and added a copy link button
- prevented the stat modal from showing up if you're not on the home page
- fixed the how to play modal bug
- other minor things i can't remember

### CODE REVIEW

**Comment by kadenbking** [comment]
@kenancasey if there's nothing major you'd like changed before you think we can deploy this on the production domain, please let me know soon. I'd like to wipe the database and merge this week. If you have minor change requests, we can make a list and I can work on those soon into next year.

**Comment by kadenbking** [comment]
@kenancasey I appreciate your changes here. I altered styling slightly. I would prefer to close your other PR -- I don't think the extra click takes too much away from the UX.

Given the amount of work I've put into this over the last few months, I would like to make a large announcement about it on social media when we merge this. Whenever you think this is good enough to go into production, I would really appreciate it if you could let me know without merging it yourself.

I would like to merge, wipe the DB, and make a FB post all in the same few minutes. Thank you for spending a small portion of your break on this. It means a lot to me!

**Comment by kenancasey** [comment]
I support that. What kind of social media announcement do you have in mind? Like through fhu's Facebook or the president's social media? I'm willing to ask though.

**Comment by kadenbking** [comment]
I was just planning on sharing it on my personal page. If the school would like to share something too that would be cool!

**Comment by kenancasey** [comment]
I'll ask Bramblett Group if they can post it to the official social media accounts. If not, you and I can post something. They might prefer to share something I make. 

Can you write up what you want it to say and share a couple of screenshots? Let's put it in a Google doc that I'm about to share. .

**Comment by kenancasey** [comment]
Take a critical look at the menu and check my design for bad UX. I made some icon choices that weren't perfect, but I think they're an improvement (over nothing). I got them all from the same library so they'd be consistent (HeroIcons - Outline variant). https://react-icons.github.io/react-icons/icons/hi/

**Comment by kadenbking** [comment]
I included a screenshot of the new menu look on the Google doc. I like them, you made good choices there! Thanks for being willing to help spread the news about this new release! I will go ahead and wipe the db, I think I have enough screenshots showcasing the new signed-in functionality. I don't want to provide too many, hopefully a small taste will encourage people to experience the new screens themselves.

**Review by kenancasey — APPROVED** [review]
I'm good with these changes. Sorry. I thought I'd already approved the PR.

File: src/components/navbar/Navbar.tsx
[the remark this answers is not in the document]
Comment (kenancasey) [reply src/components/navbar/Navbar.tsx:1]: @kadenbking Thoughts on this?

[the remark this answers is not in the document]
Comment (kadenbking) [reply src/components/navbar/Navbar.tsx:1]: Sorry, I didn't see this comment until just now. I understand/agree where you're coming from, but I don't think an additional link will fit on mobile with the current icons. Perhaps that could be a change we could place in the backlog - (re-think navbar / menu)

File: src/lib/stats.ts [hunk 1]
```diff
@@ -30,28 +56,58 @@ export const addStatsForCompletedGame = (
...
+
+  const gamesWon = gameStats.totalGames - gameStats.gamesFailed
+
+  const score =
+    gamesWon * WIN_BONUS +
+    gameStats.gamesFailed * LOSE_BONUS +
+    gameStats.successRate * SUCCESS_RATE_BONUS +
+    (6 - gameStats.avgNumGuesses) * AVG_GUESS_BONUS +
+    gameStats.currentStreak * STREAK_BONUS +
+    gameStats.bestStreak * STREAK_BONUS
```
[the remark this answers is not in the document]
Comment (kadenbking) [reply src/lib/stats.ts:110]: I agree with this as well. I have no problem with you altering the score calculation any way you see fit.

[the remark this answers is not in the document]
Comment (kenancasey) [reply src/lib/stats.ts:110]: I don't trust myself to think clearly about the interaction between all those variables but I'm going to see Jared Collins today, so I'm gonna ask him to figure it out for us. :-)

**Comment by kadenbking** [comment]
@kenancasey Happy New Year! Was Collins ever able to help you with a scoring calculation? Also, are there any updates on the FHU/Bramblett accounts being willing/able to post about v2?

**Comment by kenancasey** [comment]
Sorry, just noticed I never replied to your questions. Collins said what I wanted to do was mathematically impossible. So we can only approximate that behavior by tweaking the weights. 

Do you need my review to fix revelation?

**Comment by kadenbking** [comment]
@kenancasey no worries! That's okay, I don't think the scoring is bad anyway through the testing of myself and some of my friends/family I gave access to. 
Adding another commit (revelation) nullifies any previous approvals. You don't have to re-approve, I can forge merge on Tuesday, but the branch rules would like you to.

**Comment by kenancasey** [comment]
I'd really like to push that revelation fix out now so that we can avoid frustrating our small but passionate user base.

What are the best options for that?

**Comment by kadenbking** [comment]
I can commit straight to main then 👍

**Pull request merged**